Presentation.Web/app/components/data-processing/tabs/data-processing-agreement-tab-roles.controller.ts

```diff
@@ -0,0 +1,271 @@
+﻿module Kitos.DataProcessing.Agreement.Edit.Roles {
+    "use strict";
+
+    export class EditRolesDataProcessingAgreementController {
+        static $inject: Array<string> = [
+            "dataProcessingAgreementService",
+            "notify",
+            "hasWriteAccess",
+            "dataProcessingAgreement",
+            "dataProcessingAgreementRoles",
+            "select2LoadingService"
+        ];
+
+        constructor(
+            private readonly dataProcessingAgreementService: Services.DataProcessing.IDataProcessingAgreementService,
+            private readonly notify,
+            public hasWriteAccess,
+            private readonly dataProcessingAgreement: Models.DataProcessing.IDataProcessingAgreementDTO,
+            private readonly dataProcessingAgreementRoles: Models.DataProcessing.IDataProcessingRoleDTO[],
+            private readonly select2LoadingService: Services.ISelect2LoadingService) {
+        }
+
+        headerName = this.dataProcessingAgreement.name;
+
+        roles = _.map(this.dataProcessingAgreementRoles, role => this.mapToRoleOptions(role));
+
+        selectedRoleIdAsString = this.roles[0].id.toString();
+        selectedUser: Models.ViewModel.Generic.Select2OptionViewModel;
+
+        userOptions = this.getAvailableUserOptions(() => parseInt(this.selectedRoleIdAsString));
+
+        assignedRoles = _.orderBy(
+            _.map(this.dataProcessingAgreement.assignedRoles, assignedRole => this.mapToEditableAssignedRole(assignedRole)),
+            assignedRole => [assignedRole.role.name, assignedRole.user.name, assignedRole.user.email],
+            "asc");
+
+        lastSortedBy = "";
+        shouldSortAsc = true;
```

Comment:
du både erklærer og sætter en masse properties her men i virkeligheden kunne du erklære for sig og så initialisere i din ctor (som den jo er til for)